splunk-devops-extend/src/main/java/com/splunk/splunkjenkins/console/LabelConsoleLineStream.java

```diff
@@ -47,7 +47,7 @@ protected void eol() {
         String line = decoder.decodeLine(branch.getBuffer(), branch.size());
         // reuse the buffer under normal circumstances
         branch.reset();
-        if (StringUtils.isNotBlank(line)) {
+        if (line != null && StringUtils.isNotBlank(ANSI_COLOR_ESCAPE.matcher(line).replaceAll(""))) {
```

Comment:
regex `ANSI_COLOR_ESCAPE.matcher(line).replaceAll("")` will run twice here and it is a waste of time

Reply:
How the check should be implemented then to make sure that line is not empty after ANSI CODE replace ?